From 8fdfa962b68e2f0dc385636345f485d581322c66 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
Date: Tue, 29 Sep 2026 14:05:50 +0200
Subject: [PATCH] FROMGIT: doc: docbook2man does not accept parameter
 --encoding=UTF-8

In meson.build we find docbook2man with:

meson.build:338:
sgml2man = find_program('docbook2X2man', 'docbook2x-man', 'db2x_docbook2man', 'docbook2man', 'docbook-to-man', required: false, version: '>=0.8')

Building on Ubuntu 26.24 with `make all` results in:

    doc/meson.build:67:8: ERROR: Command `/usr/bin/docbook2man --encoding=UTF-8 lxc.sgml` failed with status 1.

build/meson-logs/meson-log.txt has:

    Running command: /usr/bin/docbook2man --encoding=UTF-8 lxc.sgml
    --- stdout ---

    --- stderr ---
    Usage: jw [<options>] <sgml_file>
    where <options> are:
      -f|--frontend <frontend>:      Specify the frontend (source format)
                                     (default is docbook)
      -b|--backend <backend>:        Specify the backend (destination format)
                                     (default is html)
      -c|--cat <file>:               Specify an extra SGML open catalog
      -n|--nostd:                    Do not use the standard SGML open catalogs
      -d|--dsl <file>|default|none:  Specify an alternate style sheet
                                     (default is to use the default stylesheet)
      -l|--dcl <file>:               Specify an alternate SGML declaration
                                     (usual ones like xml.dcl get detected automatically)
      -s|--sgmlbase <path>:          Change base directory for SGML distribution
                                     (usually /usr/share/sgml)
      -p|--parser <program>:         Specify the parser if several are installed
                                      (jade or openjade)
      -o|--output <directory>:       Set output directory
      -u|--nochunks:                 Output only one big file
                                     (overrides the stylesheet settings)
      -i|--include <section>:        Specify a SGML marked section to include
                                     (should be marked as "ignore" in the SGML text)
      -w|--warning <warning_type>|list: Control warnings or display the allowed warning types
      -e|--errors <error_type>|list: Control errors or display the allowed error types
      -h|--help:                     Print this help message and exit
      -V <variable[=value]>:         Set a variable
      -v|--version:                  Print the version and exit

Remove the illegal parameter --encoding=UTF-8.

Signed-off-by: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>

Link: https://github.com/lxc/lxc/pull/4730
Signed-off-by: Kaiyang Wu <wukaiyang@loongfans.cn>
---
 doc/ja/meson.build | 2 +-
 doc/ko/meson.build | 2 +-
 doc/meson.build    | 4 ++--
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/doc/ja/meson.build b/doc/ja/meson.build
index 449aae967..4dd8cf32a 100644
--- a/doc/ja/meson.build
+++ b/doc/ja/meson.build
@@ -67,7 +67,7 @@ if want_mans
         configure_file(
             input: sgml,
             output: page[0] + '.' + page[1],
-            command: [sgml2man, '--encoding=UTF-8', page[0] + '.sgml'],
+            command: [sgml2man, page[0] + '.sgml'],
             install: true,
             install_dir: join_paths(mandir, 'ja', 'man' + page[1]))
     endforeach
diff --git a/doc/ko/meson.build b/doc/ko/meson.build
index 80db7a2d6..295e97938 100644
--- a/doc/ko/meson.build
+++ b/doc/ko/meson.build
@@ -60,7 +60,7 @@ if want_mans
         configure_file(
             input: sgml,
             output: page[0] + '.' + page[1],
-            command: [sgml2man, '--encoding=UTF-8', page[0] + '.sgml'],
+            command: [sgml2man, page[0] + '.sgml'],
             install: true,
             install_dir: join_paths(mandir, 'ko', 'man' + page[1]))
     endforeach
diff --git a/doc/meson.build b/doc/meson.build
index 79d52fecc..8f72c95d8 100644
--- a/doc/meson.build
+++ b/doc/meson.build
@@ -67,7 +67,7 @@ if want_mans
         configure_file(
             input: sgml,
             output: page[0] + '.' + page[1],
-            command: [sgml2man, '--encoding=UTF-8', page[0] + '.sgml'],
+            command: [sgml2man, page[0] + '.sgml'],
             install: true,
             install_dir: join_paths(mandir, 'man' + page[1]))
     endforeach
@@ -84,4 +84,4 @@ if want_api_docs
         command: [doxygen, '@INPUT@'],
         build_by_default: true
     )
-endif
\ No newline at end of file
+endif
-- 
2.56.0

